### DISCUSSION TITLE
Sub Calculations

### DISCUSSION THREAD

**Discussion by gotfredsen** [kazad/instacalc#73] [Ideas]
Sometimes having one big calculation can be overwhelming, both for developer as well for the user.

What about a "function call" style sub calculator, that returns one value to the main calculation. You click on a button and it takes you to the subcalc, and when finished, it copies the result back as a value of the button, say R5

**Comment by kazad** [comment]
Thanks for the suggestion! I've wanted to add a feature where you can reference or import other calcs into the current one, similar to how you can bring in rows from a different sheet in Excel. I'll have to think about a clean way to implement this -- appreciate the idea!

**Comment by kazad** [comment]
Quick update: I have a beta feature to import one calc into another and use its variables: https://github.com/kazad/instacalc/discussions/95
